lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -217,14 +217,21 @@ func (handler *ContainersHandlersImpl) CommitHandler(params containers.CommitPar
 func (handler *ContainersHandlersImpl) RemoveContainerHandler(params containers.ContainerRemoveParams) middleware.Responder {
 	defer trace.End(trace.Begin("Containers.RemoveContainerHandler"))
 
-	//get the indicated container for removal
+	// get the indicated container for removal
 	cID := exec.ParseID(params.ID)
 	h := exec.GetContainer(cID)
 	if h == nil {
 		return containers.NewContainerRemoveNotFound()
 	}
 
-	err := h.Container.Remove(context.Background(), handler.handlerCtx.Session)
+	// make sure the container exists - it may still be cached but it may have
+	// been removed out-of-band on the backend
+	_, err := exec.ContainerInfo(context.Background(), handler.handlerCtx.Session, cID)
```

Comment:
Is this necessary or can we check the return from Remove for NotFound? There's always going to be a timing hole here in between checking it exists, calling remove and the remove actually being dispatched.